pkg/validate/container.go

```diff
@@ -318,3 +368,68 @@ func createVolumeContainer(rc internalapi.RuntimeService, ic internalapi.ImageMa
 
 	return createContainer(rc, ic, containerConfig, podID, podConfig)
 }
+
+// createLogContainer creates a container with log and the prefix of containerName.
+func createLogContainer(rc internalapi.RuntimeService, ic internalapi.ImageManagerService, prefix string, podID string, podConfig *runtimeapi.PodSandboxConfig) (string, string) {
+	By("create a container with log and name")
+	containerName := prefix + framework.NewUUID()
+	path := fmt.Sprintf("%s.log", containerName)
+	containerConfig := &runtimeapi.ContainerConfig{
+		Metadata: buildContainerMetadata(containerName, defaultAttempt),
+		Image:    &runtimeapi.ImageSpec{Image: defaultContainerImage},
+		Command:  []string{"echo", defaultLog},
+		LogPath:  path,
+	}
+	return containerConfig.LogPath, createContainer(rc, ic, containerConfig, podID, podConfig)
+}
+
+// parseDockerJSONLog parses logs in Docker JSON log format.
+// Docker JSON log format example:
+//   {"log":"content 1","stream":"stdout","time":"2016-10-20T18:39:20.57606443Z"}
+//   {"log":"content 2","stream":"stderr","time":"2016-10-20T18:39:20.57606444Z"}
+func parseDockerJSONLog(log []byte, msg *logMessage) {
+	var l jsonlog.JSONLog
+
+	err := json.Unmarshal(log, &l)
+	framework.ExpectNoError(err, "failed with %v to unmarshal log %q", err, l)
+
+	msg.timestamp = l.Created
+	msg.stream = streamType(l.Stream)
+	msg.log = []byte(l.Log)
+}
+
+// verifyLogContents verifies the contents of container log.
+//
+// CRI log format example :
+//   2016-10-06T00:17:09.669794202Z stdout The content of the log entry 1
+//   2016-10-06T00:17:10.113242941Z stderr The content of the log entry 2
+func verifyLogContents(podConfig *runtimeapi.PodSandboxConfig, logPath string, expectedLogMessage *logMessage) {
+	path := filepath.Join(podConfig.LogDirectory, logPath)
+	f, err := os.Open(path)
+	framework.ExpectNoError(err, "failed to open log file: %v", err)
+	framework.Logf("Open log file %s\n", path)
+	defer f.Close()
+
+	log, err := ioutil.ReadAll(f)
+	framework.ExpectNoError(err, "failed to read log file: %v", err)
+	framework.Logf("Log file context is %s\n", log)
+
+	var msg logMessage
+
+	// to determine whether the log is Docker format or CRI format.
+	// parse logs in Docker JSON log format
+	if strings.Contains(string(log), "{") {
+		parseDockerJSONLog(log, &msg)
+	} else { //parse logs in CRI log format
+		slog := string(log)
+
+		cstream := regexp.MustCompile(string(stdoutType))
```

Comment:
This seems not sufficient, the order is not guaranteed, and timestamp format is not checked.

You could split the string with the first 2 spaces. The first field should be timestamp, and conform to the expected timestamp format. The second field should be stream type. The third field is log content.

You can also add a new function for this.